engine/src/test/java/com/datatorrent/stram/client/AsyncTester.java

```diff
@@ -0,0 +1,61 @@
+/**
```

Comment:
Replace with the standard license header.

Reply:
Can I replace this with our license header when I de facto just copy and pasted this code from [stack overflow](https://stackoverflow.com/a/2596530)?

Reply:
Is this needed at all, can you not use ExecutorService instead? https://stackoverflow.com/a/40605786/3148138

Regarding license header, all files have to have the ASL header. As to whether you can copy the code or not:

https://meta.stackexchange.com/questions/271080/the-mit-license-clarity-on-using-code-on-stack-overflow-and-stack-exchange

> You don’t have to include the full MIT License in your code base. Contributors agree to give code users permission to ignore the MIT License’s notice preservation requirement, as long as users give reasonable attribution upon request of the copyright holder (or Stack Exchange on behalf of the contributor). This optional exception to the MIT License will live in our terms of service.